src/cdc_uart.c

```diff
@@ -37,36 +38,63 @@ TickType_t last_wake, interval = 100;
 static uint8_t tx_buf[CFG_TUD_CDC_TX_BUFSIZE];
 static uint8_t rx_buf[CFG_TUD_CDC_RX_BUFSIZE];
 
+static uint16_t cdc_debug_fifo_read;
+static uint16_t cdc_debug_fifo_write;
+static uint8_t cdc_debug_fifo[4096];
+static uint8_t cdc_debug_buf[CFG_TUD_CDC_TX_BUFSIZE];
+
+
+
 void cdc_uart_init(void) {
     gpio_set_function(PICOPROBE_UART_TX, GPIO_FUNC_UART);
     gpio_set_function(PICOPROBE_UART_RX, GPIO_FUNC_UART);
     gpio_set_pulls(PICOPROBE_UART_TX, 1, 0);
     gpio_set_pulls(PICOPROBE_UART_RX, 1, 0);
     uart_init(PICOPROBE_UART_INTERFACE, PICOPROBE_UART_BAUDRATE);
+    cdc_debug_fifo_read = 0;
+    cdc_debug_fifo_write = 0;
 }
 
 void cdc_task(void)
 {
     static int was_connected = 0;
-    uint rx_len = 0;
+    static uint32_t rx_len = 0;
 
     // Consume uart fifo regardless even if not connected
     while(uart_is_readable(PICOPROBE_UART_INTERFACE) && (rx_len < sizeof(rx_buf))) {
         rx_buf[rx_len++] = uart_getc(PICOPROBE_UART_INTERFACE);
     }
 
     if (tud_cdc_connected()) {
-        was_connected = 1;
-        int written = 0;
-        /* Implicit overflow if we don't write all the bytes to the host.
-         * Also throw away bytes if we can't write... */
-        if (rx_len) {
-          written = MIN(tud_cdc_write_available(), rx_len);
-          if (written > 0) {
-            tud_cdc_write(rx_buf, written);
-            tud_cdc_write_flush();
-          }
+      was_connected = 1;
```

Comment:
Probably better to stick to 4-spaces per indent? (and yes, the original code wasn't entirely consistent here)

Reply:
Do you suggest some reformatting?  What are the actual rules?

Reply:
I dunno if there are any hard-and-fast rules, but https://github.com/raspberrypi/pico-sdk should probably be considered as the default coding style? :shrug:

Reply:
seems to be a little bit like "WebKit".